src/librustc_privacy/lib.rs

```diff
@@ -951,18 +949,10 @@ impl<'a, 'tcx: 'a> TypeVisitor<'tcx> for SearchInterfaceForPrivateItemsVisitor<'
                     self.min_visibility = vis;
                 }
                 if !vis.is_at_least(self.required_visibility, self.tcx) {
-                    if self.tcx.sess.features.borrow().pub_restricted || self.has_old_errors {
-                        let mut err = struct_span_err!(self.tcx.sess, self.span, E0446,
-                            "private type `{}` in public interface", ty);
-                        err.span_label(self.span, &format!("can't leak private type"));
-                        err.emit();
-                    } else {
-                        self.tcx.sess.add_lint(lint::builtin::PRIVATE_IN_PUBLIC,
-                                               node_id,
-                                               self.span,
-                                               format!("private type `{}` in public \
-                                                        interface (error E0446)", ty));
-                    }
+                    let mut err = struct_span_err!(self.tcx.sess, self.span, E0446,
```

Comment:
Even for `hir::Visibility`, I still only see `Public`, `Crate`, `Restricted {...}`, and `Inherited`, no `Private`. Am I missing something obvious?